### PR TITLE
Update DiscordExceptions to include the `JsonMessage` within the `Message` property.

### PR DESCRIPTION
**Pull request by OoLunar**
# Summary
Update DiscordExceptions to include the `JsonMessage` within the `Message` property. This provides a much nicer experience when attempting to debug from a breakpoint instead of debugging by logging.

# Notes
Actually tested.

### CODE REVIEW

File: DSharpPlus/Exceptions/DiscordException.cs [hunk 1]
```diff
@@ -24,27 +24,29 @@
...
+    /// Gets the JSON message received.
+    /// </summary>
+    public virtual string? JsonMessage { get; internal set; } = null!;
 
-        /// <summary>
-        /// Gets the JSON message received.
-        /// </summary>
-        public virtual string JsonMessage { get; internal set; }
+    /// <inheritdoc />
+    public override string Message => $"{base.Message}. Json Message: {JsonMessage ?? "No Json Message Provided."}";
```
Comment (akiraveliara) [inline DSharpPlus/Exceptions/DiscordException.cs:47]: this really does not need to be in title case; and i think we should come up with a better name than JsonMessage. something like "Discord did not provide an error message."?

File: DSharpPlus/Exceptions/DiscordException.cs [hunk 2]
```diff
@@ -24,27 +24,29 @@
...
+    public virtual RestResponse WebResponse { get; internal set; } = null!;
 
-        /// <summary>
-        /// Gets the response to the request.
-        /// </summary>
-        public virtual RestResponse WebResponse { get; internal set; }
+    /// <summary>
+    /// Gets the JSON message received.
+    /// </summary>
+    public virtual string? JsonMessage { get; internal set; } = null!;
```
Comment (akiraveliara) [inline DSharpPlus/Exceptions/DiscordException.cs:44]: nit: `= null!;` is unnecessary here

**Review by akiraveliara — COMMENTED** [review]
left some notes, nothing super important though

**Comment by akiraveliara** [comment]
@OoLunar you will have to resolve merge conflicts in this branch now

**Comment by OoLunar** [comment]
Nothing a force push can't solve. With a little bit of rebasing.

**Comment by OoLunar** [comment]
Turns out `git merge -X theirs origin/master` works too.

File: DSharpPlus/Exceptions/DiscordException.cs (hunk 2 above)
Comment (OoLunar) [reply DSharpPlus/Exceptions/DiscordException.cs:44]: Compiler warning. I could readjust the constructor though.

Comment (OoLunar) [reply DSharpPlus/Exceptions/DiscordException.cs:44]: Wait nevermind this is on a nullable property

**Pull request merged**